### DISCUSSION TITLE
Flags for Behavoir

### DISCUSSION THREAD

**Discussion by Shadowblitz16** [Wohlstand/TheXTech#670] [Ideas]
I have a few suggestions for world flags that modify how the game works.

- leaf no sparkle
- leaf tap to fly
- tanooki no sparkle
- tanooki tap to fly
- 2 fireball limit
- fireball no aim
- 2 hammer limit
- hammer no aim
- hammer no sparkle
- SMB3 Jump physics
- SMB3 run physics
- SMB3 turn physics
- SMB3 swim physics
- SMB3 fly physics
- enemies turn when collided with
- tail -> upside down shell
- SMB3 style plant carry for mario and luigi
- SMB3 kick frames

**Comment by LoveBodhi** [comment]
Part of Lua script system (Gameplay feature) I think...?

**Comment by ds-sloth** [comment]
Hi, thanks for the suggestions! Yes, this kind of thing may be supported by the scripting engine in the future, but won't be included in the main game engine.

TheXTech is primarily intended for preserving the SMBX 1.3 engine logic and for research to the Moondust engine, which is planned to be more open-ended, modern, and customizable. Please see the [Project Goals](https://github.com/Wohlstand/TheXTech/wiki/Project-Goals) for more information.

I'll convert this issue into a discussion, everyone can feel free to discuss the specific ideas here, which ones are most exciting to them, etc, in the discussion.

**Comment by LoveBodhi** [comment]
More Ideas:

- SMBW Style Physics
- Hello Mario Engine style multi-direction sprout
- SMM Style Downward Vine Sprout
- Ground Pound without Purple Yoshi
- Spin-jump break for bricks instead of rotating blocks (HME Style)
- HME Style Player Physics

---

09-17 Add-on: These maybe only exclusive to the Chocolate (forked) TheXTech Engine, instead of Vanilla.

**Comment by LoveBodhi** [comment]
Anyways, no longer revelant to TheXTech engine. Closed.
